tools/IceRpc.ProtocGen/ClientGenerator.cs

```diff
@@ -18,20 +18,32 @@ internal static string GenerateInterface(ServiceDescriptor service)
             string returnType = method.OutputType.GetType(scope, method.IsServerStreaming);
             string methodName = $"{method.Name.ToPascalCase()}Async";
 
-            methods += $@"
+            if (method.GetOptions()?.Deprecated ?? false)
+            {
+                methods += @$"
+    [global::System.Obsolete]".Trim();
+            }
+            methods += @$"
     global::System.Threading.Tasks.Task<{returnType}> {methodName}(
         {inputType} {inputParam},
         IceRpc.Features.IFeatureCollection? features = null,
         global::System.Threading.CancellationToken cancellationToken = default);";
             methods += "\n";
         }
-        return @$"
+
+        string clientInterface = @$"
 /// <remarks>protoc-gen-icerpc-csharp generated this client-side interface from Protobuf service <c>{service.FullName}</c>.
-/// It's implemented by <c>{service.Name.ToPascalCase()}Client</c></remarks>
+/// It's implemented by <c>{service.Name.ToPascalCase()}Client</c></remarks>";
+        if (service.GetOptions()?.Deprecated ?? false)
+        {
+            clientInterface += "[global::System.Obsolete]";
```

Comment:
It seems we're missing a newline before this attribute.

Reply:
Added new line